src/main/java/me/botsko/prism/listeners/PrismPlayerEvents.java

```diff
@@ -39,430 +48,432 @@
+
+        RecordingQueue.addToQueue(pa);
+
+    }
+
+    /**
+     * PlayerTeleportEvent.
+     * @param event PlayerTeleportEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onPlayerTeleport(final PlayerTeleportEvent event) {
+        if (!Prism.getIgnore().event("player-teleport", event.getPlayer())) {
+            return;
+        }
+        final TeleportCause c = event.getCause();
+        if (c.equals(TeleportCause.END_PORTAL) || c.equals(TeleportCause.NETHER_PORTAL)
+                || c.equals(TeleportCause.ENDER_PEARL)) {
+            RecordingQueue.addToQueue(ActionFactory.createEntityTravel("player-teleport", event.getPlayer(),
+                    event.getFrom(), event.getTo(), event.getCause()));
+        }
+    }
+
+    /**
+     * EnchantItemEvent.
+     * @param event EnchantItemEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onEnchantItem(final EnchantItemEvent event) {
+        if (!Prism.getIgnore().event("enchant-item", event.getEnchanter())) {
+            return;
+        }
+        final Player player = event.getEnchanter();
+        RecordingQueue.addToQueue(ActionFactory.createItemStack("enchant-item", event.getItem(),
+                event.getEnchantsToAdd(), event.getEnchantBlock().getLocation(), player));
+    }
+
+    /**
+     * CraftItemEvent.
+     * @param event CraftItemEvent
+     */
+    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    public void onCraftItem(final CraftItemEvent event) {
+        final Player player = (Player) event.getWhoClicked();
+        if (!Prism.getIgnore().event("craft-item", player)) {
+            return;
+        }
+        final ItemStack item = event.getRecipe().getResult();
+        RecordingQueue.addToQueue(
+                ActionFactory.createItemStack("craft-item", item, 1, -1, null, player.getLocation(), player));
+    }
+
+    /**
+     * PlayerInteractEvent.
+     * @param event PlayerInteractEvent
+     */
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onPlayerInteract(final PlayerInteractEvent event) {
+
+        final Player player = event.getPlayer();
+        Block block = event.getClickedBlock();
+
+        ItemStack hand = player.getInventory().getItemInMainHand();
+
+        final Wand wand = Prism.playersWithActiveTools.get(player.getName());
+        // Are they using a wand (or do we always allow it)
+        if (wand != null) {
+
+            // The wand will tell us what to use.
+            final Material item_mat = wand.getItem();
+
+            // Does the player have such item?
+            if (hand.getType() == item_mat) {
+
+                // Left click is for current block
+                if (event.getAction() == Action.LEFT_CLICK_BLOCK && event.getHand() == EquipmentSlot.HAND && block != null) {
+                        wand.playerLeftClick(player, block.getLocation());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['if' child has incorrect indentation level 24, expected level should be 20.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)